src/main/scala/is/hail/expr/Fun.scala

```diff
@@ -14,6 +16,28 @@ case class Transformation[T, U](f: T => U, fcode: Code[T] => CM[Code[U]]) extend
   def apply(t: T): U = f(t)
 }
 
+case class UnaryDependentFunCode[T, U](retType: Type, code: () => Code[T] => CM[Code[U]]) extends Fun {
+  private var postCapture: Code[T] => CM[Code[U]] = null
+
+  override def captureType() {
+    postCapture = code()
```

Comment:
Aww, I liked the capture trick from before of returning a UnaryFunCode.  Definitely seems more elegant and doesn't duplicate the logic below like `subst` and `convertArgs`.